src/NUnitEngine/nunit.engine.core/Internal/TestAssemblyResolver.cs

```diff
@@ -64,8 +77,74 @@ private Assembly OnResolving(AssemblyLoadContext context, AssemblyName name)
                 }
             }
 
+            foreach(string frameworkDirectory in AdditionalFrameworkDirectories)
+            {
+                var versionDir = FindBestVersionDir(frameworkDirectory, name.Version);
+                if (versionDir != null)
+                {
+                    string candidate = Path.Combine(frameworkDirectory, versionDir, name.Name + ".dll");
+                    if (File.Exists(candidate))
+                        return _loadContext.LoadFromAssemblyPath(candidate);
+                }
+            }
+
             return null;
         }
+
+        private static string GetDotNetInstallDirectory()
+        {
+            if (Path.DirectorySeparatorChar == '\\')
+            {
+                // Running on Windows so use registry
+                RegistryKey key = Registry.LocalMachine.OpenSubKey(@"Software\dotnet\SetUp\InstalledVersions\x64\sharedHost\");
+                return (string)key?.GetValue("Path");
+            }
+            else
+                return "/usr/shared/dotnet/";
```

Comment:
What if the user have installed dotnet in another location

Reply:
In that case, it can't be found. Short of a global search of the hard drive, there seems to be no built-in way of finding installations. I believe this is actually a feature, allowing you to install a test copy of the runtime without polluting the version in the standard location.